### PR TITLE
Track files replaced by move

### PR DESCRIPTION
**Pull request by dougkoch**
Adds a watch back to any file that has been replaced by a move. We look for a fsnotify.Remove event and retry watcher.Add() until it returns an error code of 0. Once a max number of retries has been exceeded, it is assumed that the file was removed.

Fixes: #9

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by mprobst** [comment]
Alternative approach: run the build, then drop and re-add all watches. This usually gives enough time for the file to re-appear.

**Comment by dougkoch** [comment]
@mprobst Turns out that a file overwritten by a mv re-appears quickly enough that a watch can be added as soon as the remove event is received. My retry loop (removed in my latest commit) was unnecessary.
@achew22 Ready for review

File: ibazel/main.go [hunk 1]
```diff
@@ -59,6 +59,25 @@ ibazel build //path/to/my/buildable:target
 `)
 }
 
+type SourceEventProcessor struct {
```
Comment (achew22) [inline ibazel/main.go]: Can you move this into its own file?

File: ibazel/main.go [hunk 2]
```diff
@@ -59,6 +59,25 @@ ibazel build //path/to/my/buildable:target
...
+	SourceFileEvents chan fsnotify.Event
+	SourceFileWatcher *fsnotify.Watcher
+}
+
+func (s SourceEventProcessor) Listen() {
```
Comment (achew22) [inline ibazel/main.go]: Can you add a method that creates a SourceEventProcessor and creates the channel then calls

go s.Listen()

File: ibazel/main.go [hunk 3]
```diff
@@ -107,7 +132,7 @@ func main() {
 		switch state {
 		case WAIT:
 			select {
-			case <-sourceFileWatcher.Events:
+			case <-sourceFileEvents:
```
Comment (achew22) [inline ibazel/main.go]: Can you refer to the channel as being attached to the object instead of in the function scope?

File: ibazel/main.go (hunk 1 above)
Comment (dougkoch) [inline ibazel/main.go]: Done

File: ibazel/main.go (hunk 2 above)
Comment (dougkoch) [inline ibazel/main.go]: Done

File: ibazel/main.go (hunk 3 above)
Comment (dougkoch) [inline ibazel/main.go]: Done

**Pull request merged**